lib/rspec/matchers/built_in/change.rb

```diff
@@ -371,6 +371,14 @@ def evaluate_value_proc
           end
         end
 
+        def inspect_message_with_receiver(message, receiver)
+          if receiver.class == Class
+            "#{receiver}.#{message}"
+          else
+            "#{receiver.class}##{message}"
```

Comment:
We also need to handle objects that don't respond to `#class` here like the following:

```ruby
singleton_class = class << receiver; self; end
singleton_class.ancestors.find { |ancestor| !ancestor.equal?(singleton_class) }
```

https://github.com/rspec/rspec-support/blob/70a8b33fb194f5014e37d99dc3d9acca88ab8647/lib/rspec/support/object_formatter.rb#L202-L203